controllers/local_db_services.go

```diff
@@ -72,18 +72,20 @@ func (r *BackstageReconciler) reconcileLocalDbServices(ctx context.Context, back
 
 }
 
-func (r *BackstageReconciler) reconcilePsqlService(ctx context.Context, backstage bs.Backstage, serviceName, label, configKey, ns string) error {
+func (r *BackstageReconciler) reconcilePsqlService(ctx context.Context, backstage *bs.Backstage, serviceName, label, configKey, ns string) error {
 	service := &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      serviceName,
 			Namespace: ns,
 		},
 	}
-	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, service, r.serviceObjectMutFun(ctx, service, backstage, backstage.Spec.RawRuntimeConfig.LocalDbConfigName, configKey, serviceName, label)); err != nil {
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, service, r.serviceObjectMutFun(ctx, service, *backstage, backstage.Spec.RawRuntimeConfig.LocalDbConfigName, configKey, serviceName, label)); err != nil {
 		if errors.IsConflict(err) {
 			return fmt.Errorf("retry sync needed: %v", err)
 		}
-		return err
+		msg := fmt.Sprintf("failed to sync database service: %s", err.Error())
+		setStatusCondition(backstage, bs.ConditionSynced, metav1.ConditionFalse, bs.SyncFailed, msg)
```

Comment:
Why do we need 2 conditions making the same status and message on both?

Reply:
You are right, the error messages are duplicated. The two conditions are used as one is to indicate whether localdb has been successfully deployed while the other is indicate whether all resources have been deployed.

Reply:
Yes, I see that, and my suggestion is to have only one with informative enough message.

Reply:
This sounds good to me. I removed the RouteSynced and LocalDBSynced conditions from the status. We can add further enhancements in the future if needed.